scripts/qemu/ia32-generic.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#
+# Shell script for running Phoenix-RTOS on QEMU (ia32-generic)
+#
+# Copyright 2021 Phoenix Systems
+# Author: Lukasz Kosinski
+#
+
+CMD="qemu-system-i386"
+
+if [ "$#" -eq 0 ]; then
+	# No args (default command)
+	CMD+=" -hda ../../_boot/phoenix-ia32-generic.disk"
+else
+	hd=0
+	vblk=0
+	# Parse args
+	while [ "$#" -gt 0 ]; do
+		case "$1" in
+			# Load kernel (.elf)
+			-kernel)
+				CMD+=" -kernel $2"
+				shift 2
+				;;
+			# Redirect serial port
+			-serial)
+				CMD+=" -serial $2"
+				shift 2
+				;;
+			# Add ATA drive
+			-ata|-ide)
+				CMD+=" -drive file=$2,if=none,id=hd$hd -device ide-hd,drive=hd$((hd++))"
+				shift 2
+				;;
+			# Add VirtIO drive (modern)
+			-vblk|-virtio-blk)
+				CMD+=" -drive file=$2,cache=unsafe,if=none,id=vblk$vblk -device virtio-blk-pci,disable-legacy=on,disable-modern=off,drive=vblk$((vblk++))"
+				shift 2
+				;;
+			# Add VirtIO GPU
+			-vgpu|-virtio-gpu)
+				CMD+=" -device virtio-gpu-pci"
+				shift
+				;;
+			*)
+				echo "$1: unknown option"
+				exit 1
+				;;
+		esac
+	done
+fi
+
+# Print and execute command
+echo "$CMD" && eval "$CMD"
```

Comment:
maybe exec would be better not to fork from script but directly execute qemu?

Reply:
Done. Good idea since the scripts are just wrappers for qemu startup command.